precompiles/proxy/src/lib.rs

```diff
@@ -161,6 +193,67 @@ where
 		Ok(())
 	}
 
+	/// Dispatch the given subcall (`call_to`, `call_data`) from an account that the sender is
+	/// authorised for through `add_proxy`.
+	///
+	/// Parameters:
+	/// - `real`: The account that the proxy will make a call on behalf of.
+	/// - `force_proxy_type`: Specify the exact proxy type to be used and checked for this call
+	/// (optional parameter, use `255` for None).
+	/// - `call_to`: Recipient of the call to be made by the `real` account.
+	/// - `call_data`: Data of the call to be made by the `real` account.
+	#[precompile::public("proxy(address,uint8,address,bytes)")]
+	fn proxy(
+		handle: &mut impl PrecompileHandle,
+		real: Address,
+		force_proxy_type: u8,
+		call_to: Address,
+		call_data: BoundedBytes<GetCallDataLimit>,
+	) -> EvmResult {
+		let real_account_id = Runtime::AddressMapping::into_account_id(real.clone().into());
+		let who = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		let force_proxy_type = match force_proxy_type {
+			255 => None,
```

Comment:
Is this a common solidity `Option` pattern? Think previously it was also discussed to perhaps have two methods `proxy` & `proxy_force_type` to get around the `Option` limitation.

Reply:
Done